tracer/src/Datadog.Trace/Telemetry/Collectors/MetricsTelemetryCollector.cs

```diff
@@ -210,6 +221,24 @@ public void Clear()
         return data;
     }
 
+    private string[]? GetTags(string? ns, string[]? metricKeyTags)
+    {
+        if (ns != MetricNamespaceConstants.ASM)
+        {
+            return metricKeyTags;
+        }
+
+        if (metricKeyTags is null)
+        {
+            return _wafVersionTags ?? _unknownWafVersionTags;
+        }
+
+        var wafVersionTag = (_wafVersionTags ?? _unknownWafVersionTags)[0];
+
+        metricKeyTags[0] = wafVersionTag;
```

Comment:
We should add a comment somewhere saying that the wafversion tag should always be the first one on asm metrics. Edit: adding a related suggestion in MetricTags